srm_trainer.py

```diff
@@ -0,0 +1,78 @@
+"""
+Command line interface to run the neural network model!
+
+From the project root directory, do:
+
+    python trainer.py fit
+
+References:
+- https://lightning.ai/docs/pytorch/2.1.0/cli/lightning_cli.html
+- https://pytorch-lightning.medium.com/introducing-lightningcli-v2-supercharge-your-training-c070d43c7dd6
+"""
+import lightning as L
+from lightning.pytorch.callbacks import ModelCheckpoint
+from lightning.pytorch.cli import ArgsType, LightningCLI
+from lightning.pytorch.loggers import CSVLogger
+from lightning.pytorch.plugins.io import AsyncCheckpointIO
+
+from src.model_geomae import GeoMAEModule
+from src.srm_datamodule import ClayDataModule
+
+
+def main():
+    dm = ClayDataModule(batch_size=128, num_workers=8)
+    model = GeoMAEModule(lr=1e-2)
+    trainer = L.Trainer(
+        fast_dev_run=False,
+        accelerator="gpu",
+        devices=1,
+        # precision="16-mixed",
+        max_epochs=50,
+        # log_every_n_steps=100,
+        logger=CSVLogger("logs", name="geomae"),
+    )
+
+    trainer.fit(model, dm)
```

Comment:
Let me know once you've got the code to a stable state, and I can help with setting up the LightningCLI code to allow for different datamodules/models, following https://lightning.ai/docs/pytorch/2.1.0/cli/lightning_cli_intermediate_2.html#multiple-lightningdatamodules. That way we can have a single trainer file and do:

```
python trainer.py fit --data=GeoTIFFDataPipeModule --model=ViTLitModule  # from PR52
python trainer.py fit --data=ClayDataModule --model=GeoMAEModule  # this PR47
```

Reply:
Sure, I initially had issues with the cuda 12.x set up on my machine & thought it might be because of Lightning CLI.
Yes, I would like to move to CLI once the experiment is working fine.

Reply:
Is it an error like `RuntimeError: GET was unable to find an engine to execute this computation`? I've been getting that on my AWS EC2 G5 instance, and had to set `export TORCH_CUDNN_V8_API_DISABLED=1` as a workaround from https://discuss.pytorch.org/t/segment-anything-model-unable-to-find-engine-to-execute-this-computation/179172.